### PR TITLE
search: Notepad updates

### PR DESCRIPTION
**Pull request by fkling**
**Note**: I have to update the tests, but I wanted to share what I already have.
**Note 2:** The screenshots still show the pencil icon, but I have changed the implementation to use the pen icon already.

This PR implements the discussed visual changes for the closed and opened Notepad state. In particular the closed state is now much simpler, simply showing a button to open the notepad. The opened state now contains a more extensive "add note/entry" section.

Some additional tweaks:

- There is no only one open/close button instead of two (the open header used to be two buttons...)
- Meta+Enter in the annotations input will now close it.
- I choose slightly lighter backgrounds for the selected entries in light mode. I think the previous background color was too dark/heavy.
- Changed the heading to `<h2>` since it isn't actually contained in any other context.

## Test plan

TBD

## App preview:

- [Link](https://sg-web-fkling-notepad-minimized-state.onrender.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/search/SearchStack.tsx [hunk 1]
```diff
@@ -270,146 +278,124 @@ export const SearchStack: React.FunctionComponent<SearchStackProps> = ({ initial
...
-                                    focus={hasNewEntry && index === 0}
-                                    selected={selected}
-                                    onDelete={selected ? deleteSelectedEntries : deleteEntry}
-                                />
-                            </li>
-                        )
-                    })}
-                </ul>
-                {confirmRemoveAll && (
+    const buttonLabel = (open ? 'Open' : 'Close') + ' Notebook'
```
Comment (novoselrok) [inline client/web/src/search/SearchStack.tsx:281]: Should this be Open/Close Notepad, not Notebook?

**Review by novoselrok — APPROVED** [review]
One small comment, otherwise 🔥.

File: client/web/src/search/SearchStack.tsx (hunk 1 above)
Comment (fkling) [reply client/web/src/search/SearchStack.tsx:281]: Of course :D Thanks for catching that!

**Comment by benvenker** [comment]
@fkling this looks really good! Question: what happens to the “save as” button on hover (if anything)? I’m wondering if we should match its color to the primary background color of the notepad component in each theme so its more visible. What do you think? I’m also a little worried that design will take issue with the blue outline, but only because I don’t recall seeing that button style in our design system.

**Comment by fkling** [comment]
@benvenker It's our default button component (see Outline): https://storybook.sgdev.org/?path=/story/wildcard-button--all-buttons . It's transparent by design (so it has the same background-color as the containing element). Hovering over it changes the background color to white (or some other color in dark mode I guess :D).

**Comment by benvenker** [comment]
Ok cool!

**Review by benvenker — APPROVED** [review]

**Pull request merged**